### DISCUSSION TITLE
Video prefetch and store in cache

### DISCUSSION THREAD

**Discussion by slavko-lukic** [TheWidlarzGroup/react-native-video#4964] [Ideas]
### What problem are you trying to solve?

Depends a lot on app use case, but in some apps you maybe don't want to stop the flow while video downloads. Prefetching a video and storing it in cache would ensure that there will be no buffering/loading delay later when you need it.

### Proposed solution

A function that will allow prefetching video similarly like you can prefetch images. The prefetched video will be stored in cache.

I imagine it would work something like this.

```javascript
Video.prefetch("someUrl").then((isSuccess) => {
    // do something
})

// or

const isSuccess = await Video.prefetch("someUrl");
```

Param could be string or array of strings if you want to prefetch more than one url.

**Comment by freeboub** [comment]
Good idea, but more props are needed. I think similar props than the source would be better. If possible should be something similar to the download to go feature

**Comment by yash-vaghela21** [comment]
@slavko-lukic This helps for now - https://github.com/zhigang1992/react-native-video-cache#readme
not prefetching but cache the video once it played.

**Comment by derynusaputra** [comment]
> @slavko-lukic This helps for now - https://github.com/zhigang1992/react-native-video-cache#readme not prefetching but cache the video once it played.

I have tried this method, but it doesn't work, can you help me? I'm making a TikTok-like project using hls video files. The problem is that when you swipe down, first start the video loading first.

**Comment by gideon-uiby** [comment]
Any update on the feature request?

**Comment by hari-mohan-choudhary** [comment]
@gideon-uiby did you able to find any solution. I was looking same solution

**Comment by freeboub** [comment]
No progress on this point, it is an heavy task to implement (especially for dash and m3u). maybe you can join TheWidlzarGroup to discuss about this subject !

**Comment by TWG-AI** [comment]
v7 supports pre-loading via `player.preload()`, which initializes the native player and starts buffering before the view is shown:

```ts
const player = useVideoPlayer(source);
await player.preload(); // buffer ahead of time
// ...later
<VideoView player={player} />
```

For TikTok-style feeds, `preload()` + `replaceSourceAsync()` is the recommended v7 pattern for instant source switching.

> I'm an AI — see the v7 docs for the full preloading flow.
